src/screens/DiscoveryScreenV2/index.js

```diff
@@ -227,7 +224,7 @@ const DiscoveryScreenV2 = ({route}) => {
     cancelTokenRef.current = axios.CancelToken.source();
   };
 
-  const renderFragment = () => {
+  const RenderFragment = React.useMemo(() => {
     if (selectedScreen === DISCOVERY_TAB_USERS)
       return (
         <UsersFragment
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [227-294]

The conversion of `renderFragment` into a memoized component `RenderFragment` using `React.useMemo` is a good practice for optimizing rendering performance. However, please ensure that the dependencies array `[selectedScreen]` includes all variables and props that the component depends on to prevent unnecessary re-renders or stale data.
```diff
-  const renderFragment = () => {
+  const RenderFragment = React.useMemo(() => {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->